src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -61,19 +64,42 @@ class TransportReceiver
 
         async Task InvokePipeline(PushContext pushContext)
         {
+            //note: we should get this value from the transport in the future to make it more accurate
+            var pipelineStartedAt = DateTime.UtcNow;
+
+
             using (var childBuilder = builder.CreateChildBuilder())
             {
                 var rootContext = new RootContext(childBuilder, pipelineCache, eventAggregator);
-                var context = new TransportReceiveContext(new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream), pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+
+                //notifications are only relevant for the main pipeline since satellites pipelines are not exposed to user in any way
+                if (isMainReceiver)
+                {
+                    var sw = new Stopwatch();
+
+                    sw.Start();
+
+                    // float the sw via the context to avoid threading issues should the continuation where we access 
+                    // it is executed on a different thread
+                    rootContext.Set(sw);
```

Comment:
@timbussmann I think we need to do more digging. @andreasohlund started using the `StopWatch` because `DateTime.UtcNow` can have issues with clock skew and then the critical time will be way off. But now we are using the `StopWatch` which is not threadsafe when you use the high precision counter because of 

http://stackoverflow.com/questions/17505807/stopwatch-c-sharp-behave-differently-in-different-threads/17505960#17505960
and see http://joeduffyblog.com/2006/03/25/a-stopwatch-for-threads/

But here comes the caveat. If I understand the code correctly we are actually not using the high resolution stop watch settings. But since it is an application wide settings a user could turn it on. If we would assume that nobody actually turns it on then [`GetTimeStamp`](http://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Stopwatch.cs,136) would use [`DateTime.UtcNow.Ticks`](http://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Stopwatch.cs,142) which means we would suffer from the same clock skew again. Or is my thought process completely off here? // cc @Scooletz @SimonCropp